commons/src/test/java/org/eclipse/kapua/commons/util/KapuaExceptionUtilsTest.java

```diff
@@ -52,16 +60,114 @@ public void testConvertPersistenceException() {
         Throwable a = new RollbackException("rollback2", r);
         PersistenceException persistExc1 = new PersistenceException(a);
 
-        Exception[] listOfPermittedExceptions = new Exception[]{kapExc0,kapExc1,kapExc2,kapExc3,
-                kapExc4,persistExc0, persistExc1,optimisticLock};
-        int sizeOfExceptions = listOfPermittedExceptions.length;
-        // positive tests
-        for (int i = 0; i < sizeOfExceptions; i++) {
+        Exception[] listOfExceptions = new Exception[]{kapExc0, kapExc1, kapExc2, kapExc3, kapExc4, persistExc0, persistExc1, optimisticLock};
+        Exception[] listOfExpectedExceptions = new Exception[]{kapExc0, kapExc1, kapExc2, kapExc3, KapuaException.internalError(kapExc4, "Error during Persistence Operation"), KapuaException.internalError(persistExc0, "Error during Persistence Operation"), KapuaException.internalError(persistExc1, "Error during Persistence Operation"), new KapuaOptimisticLockingException(optimisticLock)};
+
+        for (int i = 0; i < listOfExceptions.length; i++) {
+            try {
+                assertEquals(listOfExpectedExceptions[i].toString(), KapuaExceptionUtils.convertPersistenceException(listOfExceptions[i]).toString());
```

Comment:
assertEquals has an overload method that accepts a message as a first argument. So fail invoked in the catch exception is not the correct usage of Assert stuff.
Please change this and also other similar usages.